matter_server/common/helpers/util.py

```diff
@@ -107,7 +107,7 @@ def parse_value(
     value: Any,
     value_type: Any,
     default: Any = MISSING,
-    allow_none: bool = True,
+    allow_none: bool = False,
```

Comment:
Seems that with this PR `allow_none` is not really used anymore? Can we drop it completely?

Reply:
II couldnt spot where it was used but was scared to remove it

Reply:
it is still in use (within dataclass_from_dict)

Reply:
Hm, I see we pass it from `dataclass_from_dict()` to `parse_value()`. But since `dataclass_from_dict()` is in our code too, we can just drop it there no? It doesn't serve any purpose and might only lead to confusion. :man_shrugging:

Reply:
No, dataclass_from_dict is used in HA we cant just drop that - it does serve a purpose, parse it strict or not strict.

In strict mode a None value is seen as error if that is not in the annotations.

Reply:
Sorry, I didn't meant to suggest to drop `dataclass_from_dict`. I am suggesting to drop `allow_none` from parse_value()`. It needs a `dataclass_from_dict` update, but it should be a simple oneline change:


```patch
--- a/matter_server/common/helpers/util.py
+++ b/matter_server/common/helpers/util.py
@@ -272,7 +272,6 @@ def dataclass_from_dict(
                 dict_obj.get(field.name),
                 type_hints[field.name],
                 field.default,
-                allow_none=not strict,
                 allow_sdk_types=allow_sdk_types,
             )
             for field in dc_fields
```

Reply:
No, because that is exactly where we use it!
if the dataclass_from_dict is strict we want it to raise on None values

Reply:
Almost weekend ;-)